third_party/grpc-rust/grpc-compiler/src/codegen.rs

```diff
@@ -191,9 +245,13 @@ impl<'a> ServiceGen<'a> {
         }
     }
 
-    // trait name
+    // trait name for server functionality
     fn server_intf_name(&self) -> String {
-        format!("{}Node", self.proto.get_name())
+        if self.proto.get_name().ends_with("Service") {
+            self.proto.get_name().to_string()
+        } else {
+            format!("{}Service", self.proto.get_name())
+        }
```

Comment:
The aim was to avoid generating names like `FormatServiceService`; added a comment to that effect.

But happy to live with ugly/stuttering names and be consistent/predictable if you prefer.

Reply:
Yeah I understand the reasoning behind it, but my point is that we should rename the service in the proto and remove the unnecessary suffix from there:

https://github.com/project-oak/oak/blob/e6f0aa8611de9ca2c54f22a4b2bcc78e72dd5715/examples/abitest/proto/abitest.proto#L39

Reply:
*and* also perhaps don't try and add the `Service` suffix in the generated code (but this one I feel less strongly, so up to you -- but at least either always add it or never add it).

My personal preference would be:

```proto
service Chat { ... }
```

generates

```rust
trait Chat { ... }
// alternatively
trait ChatService { ... }

struct ChatClient { ... }
```

Reply:
The [protobuf style guide](https://developers.google.com/protocol-buffers/docs/style#services) uses an example with a "Service" suffix, so I think they're bound to keep cropping up.  So I'll drop the `Service` suffix in the generated `trait`.